tests/test_sockets.py

```diff
@@ -1849,3 +1852,41 @@ async def test_connect_tcp_getaddrinfo_context() -> None:
             pass
 
     assert exc_info.value.__context__ is None
+
+
+@pytest.mark.parametrize("socket_type", ["socket", "fd"])
+@pytest.mark.parametrize("event", ["readable", "writable"])
+async def test_wait_socket(
+    anyio_backend_name: str, event: str, socket_type: str
+) -> None:
+    if anyio_backend_name == "asyncio" and sys.platform == "win32":
+        import asyncio
+
+        policy = asyncio.get_event_loop_policy()
+        if policy.__class__.__name__ == "WindowsProactorEventLoopPolicy":
+            pytest.skip("Does not work on asyncio/Windows/ProactorEventLoop")
+
+    wait_socket = wait_socket_readable if event == "readable" else wait_socket_writable
+
+    def client(port: int) -> None:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.connect(("127.0.0.1", port))
+            sock.sendall(b"Hello, world")
+
+    with move_on_after(0.1):
```

Comment:
This makes the test flaky; let's give it more breathing room.
```suggestion
    with move_on_after(10):
```
Also, move it closer to `await wait_socket(...)` as it's the only place where it has any effect. And maybe change to `fail_after()` so that you don't need to set those flags.